src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java

```diff
@@ -348,6 +349,31 @@ public static org.cyclonedx.model.Metadata createMetadata(final Project project)
                 });
                 cycloneComponent.setExternalReferences(references);
             }
+            /*Issue #2737:  Adding supplier contact functionality */
+            if (project.getSupplier() != null) {
+                OrganizationalEntity supplier = new OrganizationalEntity();
+                supplier.setName(project.getSupplier().getName());
+            
+                if (project.getSupplier().getUrls() != null) {
+                    supplier.setUrls(project.getSupplier().getUrls());
+                } else {
+                    supplier.setUrls(null);
+                }
+                if (project.getSupplier().getContacts() != null) {
+                    List<OrganizationalContact> contacts = new ArrayList<>();
+                    for (org.cyclonedx.model.OrganizationalContact OrganizationalContact: project.getSupplier().getContacts()) {
+                        OrganizationalContact contact = new OrganizationalContact();
+                        contact.setName(OrganizationalContact.getName());
+                        contact.setEmail(OrganizationalContact.getEmail());
+                        contact.setPhone(OrganizationalContact.getPhone());
+                        contacts.add(contact);
+                    }
+                    supplier.setContacts(contacts);
+                }
+                project.setSupplier(supplier);
```

Comment:
`createMetadata` is converting *from* DT's data model *to* CycloneDX, so it should not set anything on DT's `project`:

```suggestion
                cycloneComponent.setSupplier(supplier);
```

Reply:
made the change.  will commit shortly :)

Reply:
when i make this change it is complaining for some reason 😭